fabtests/common/shared.c

```diff
@@ -563,7 +563,8 @@ int ft_alloc_msgs(void)
 		if (ret)
 			return ret;
 
-		ret = ft_alloc_host_tx_buf(MAX(tx_size, FT_MAX_CTRL_MSG) * opts.window_size);
+		size_t host_tx_size =  (opts.options & FT_OPT_ALLOC_MULT_MR) ? tx_mr_size : tx_size;
+		ret = ft_alloc_host_tx_buf(MAX(host_tx_size, FT_MAX_CTRL_MSG) * opts.window_size);
```

Comment:
Please declare variables at the top of functions, so that they can always be quickly located.  Check line lengths and limit to about 80 characters.

As for the changes, this function is becoming a mess with repeated checks.  Just in these 2 lines we have 2 branches, both of which are somewhat duplicated by earlier checks (lines 529-540) and later checks (lines 576-581).  We need to start separating some of the functionality out and make things more consistent and easier to understand.

For example, why isn't buf_size used here?  Isn't that the buffer size to allocate?  Why the host buffer is a different size isn't clear at all.

The ft_alloc_host_tx_buf() call should go away completely.  It's a 1-line function called from 2 places that only obscures what is changing.

Can we get a function that returns the size of the buffer(s) to allocate, size of that reserved for tx, size reserved for rx, etc.?

Reply:
> Please declare variables at the top of functions, so that they can always be quickly located. Check line lengths and limit to about 80 characters.

Fixed.

> For example, why isn't buf_size used here? Isn't that the buffer size to allocate? Why the host buffer is a different size isn't clear at all.

Explained in comment.

>Can we get a function that returns the size of the buffer(s) to allocate, size of that reserved for tx, size reserved for rx, etc.?

I am not sure I understand the ask. Do you want a function that set all the size, like tx_size, rx_size, tx_mr_size, rx_mr_size, buf_size, etc?